extension/goext/error_test.go

```diff
@@ -45,6 +46,10 @@ var _ = Describe("Error", func() {
 			errDouble = goext.NewError(http.StatusServiceUnavailable, goext.NewError(http.StatusUseProxy, fmt.Errorf("nothing here")))
 			errDouble.Origin = "one.go:1000"
 			errDouble.Err.(*goext.Error).Origin = "two.go:2000"
+
+			var nilErr *goext.Error = nil
+			errDoubleNil = goext.NewError(http.StatusInternalServerError, nilErr)
```

Comment:
this looks very strange. why do we allow wrapping a nil error with an error code? what's the use case?

Reply:
We don't, but we can't defend from default value of goext.Error. I've changed testcase.